pkg/envoy/agent.go

```diff
@@ -179,6 +179,10 @@ func (a *Agent) terminate() {
 				log.Infof("Envoy exited with status %v", status.err)
 				log.Infof("Graceful termination logic ended prematurely, envoy process terminated early")
 				return
+			case <-time.After(a.terminationDrainDuration - a.minDrainDuration):
```

Comment:
Given how the exitOnZeroActiveConnections feature is documented, this upperbound looks like something that should have been on master (and previous releases) as well. Do you think it makes sense to backport this part?

Reply:
Sorry. I will revert this change. Need to think about this a bit more. This is not needed for quitquitquit change

Reply:
But I think it is good to have some upper bound. Need to think about it more.